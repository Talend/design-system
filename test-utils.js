import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from './src/components/ThemeProvider';
import theme from './src/themes/light.theme';

const AllTheProviders = ({ children }) => {
	return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
