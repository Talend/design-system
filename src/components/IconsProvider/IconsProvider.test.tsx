import React from 'react';
import { render } from '../../../test-utils';
import { IconsProvider } from './index';

describe('IconsProvider', () => {
	test('should match snapshot', () => {
		const { container } = render(<IconsProvider />);
		expect(container.firstChild).toMatchSnapshot();
	});

	test('should prevent multiple instance creation', () => {
		const { container } = render(
			<>
				<IconsProvider />
				<IconsProvider />
			</>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
