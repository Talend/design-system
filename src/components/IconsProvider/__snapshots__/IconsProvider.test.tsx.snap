// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsProvider should match snapshot 1`] = `
<svg
  class="tc-iconsprovider sr-only"
  focusable="false"
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`IconsProvider should prevent multiple instance creation 1`] = `
<svg
  class="tc-iconsprovider sr-only"
  focusable="false"
  xmlns="http://www.w3.org/2000/svg"
/>
`;
